src/openforms/submissions/tests/test_submission_report.py

```diff
@@ -98,6 +100,167 @@ def test_report_generation(self, mock_request):
         self.assertTrue(report.content.name.endswith("Test_Form.pdf"))
 
     def test_report_is_generated_in_same_language_as_submission(self):
+        # fixture_data
+        fields = [
+            # Component.type, user_input
+            ("bsn", "111222333"),
+            ("checkbox", "yes"),
+            ("currency", "1100000"),
+            ("date", "1911-06-29"),
+            ("email", "hostmaster@example.org"),
+            ("file", "download(2).pdf"),
+            ("iban", "NL56 INGB 0705 0051 00"),
+            ("licenseplate", "AA-00-13"),
+            ("map", "52.193459, 5.279538"),
+            ("number", "1"),
+            ("password", "Panda1911!"),  # XXX Why is this widget even an option?
+            ("phonenumber", "+49 1234 567 890"),
+            ("postcode", "3744 AA"),
+            ("radio", "Radio number one"),
+            ("select", "A fine selection"),
+            ("selectboxes", "This; That; The Other"),
+            ("signature", SIGNATURE),
+            ("textarea", "Largish predetermined ASCII"),
+            ("textfield", "Short predetermined ASCII"),
+            ("time", "23:50"),
+            ("updatenote", "C#"),
+        ]
+        submission_data = {}
+        translations = {}
+        components = []
+
+        # carthesian products are big, let's loop to fill these
+        for component_type, user_input in fields:
+            # common base
+            component = {
+                "type": component_type,
+                "key": f"{component_type}-key",
+                "label": f"Untranslated {component_type.title()} label",
+                "showInPDF": True,
+                "hidden": False,
+            }
+            submission_data[f"{component_type}-key"] = user_input
+            translations[
+                f"Untranslated {component_type.title()} label"
+            ] = f"Translated {component_type.title()} label"
+
+            # add the component
+            components.append(component)
+
+            # but massage the weirder components into shape
```

Comment:
I'm envisioning this a bit more like:

```py


components_and_data = (
    (BSNComponentFactory, "bsn", {"extra": "bsn factory kwargs"}, 111222333),
    (CheckboxComponentFactory, "checkbox", {}, True},
    ...
)
for factory_cls, c_type, extra, user_input in components_and_data:
    component = factory_cls(key=c_type, label=f"Untranslated {c_type} label", showInPDF=True, hidden=false, **extra)
    submission_data[c_type] = user_input

```

so that you can:

* encapsulate component-specific options/configs per factory (while sharing the common attributes
* keep all factory-specifics on a single item
* you can rely on factory details to fill out some irrelevant datastructures while keeping the component definition representative
* simplifies the loop here used and does away with the `match` statement

Also note that checkboxes end up as true/false values, and the user input for a select for example is not the label, but the underlying value. You can review `DefaultFormatterTestCase` which uses json files for definitions _and_ user input values that were created by actually submitting the formio forms :)